lib/Net/Async/Blockchain/Client/Websocket.pm

```diff
@@ -101,8 +101,12 @@ sub websocket_client : method {
                     my (undef, $frame) = @_;
                     $self->source->emit(decode_json_utf8($frame));
                 },
+                on_closed => sub {
+                    die "Connection closed by peer";
+                },
             ));
 
+        $client->{close_on_read_eof} = 1;
```

Comment:
Better to do this by setting it as a constructor parameter rather than reaching directly into the object.

```perl
my $client = Net::Async::WebSocket::Client->new(
    ...
    close_on_read_eof => 1,
);
```